pkg/subctl/cmd/gather.go

```diff
@@ -190,13 +186,34 @@ func gatherDiscovery(dataType string, info *gather.Info) bool {
 }
 
 func gatherBroker(dataType string, info *gather.Info) bool {
+	brokerRestConfig, err := getBrokerRestConfig(info.Submariner)
+	exitOnError("Error getting Broker rest config", err)
+
+	dynClient, clientSet, err := getClients(brokerRestConfig)
+	exitOnError("Error getting client %s", err)
+
+	info = &gather.Info{
+		Clients:     gather.Clients{ClientSet:clientSet, DynClient:dynClient},
+		RestConfig:  brokerRestConfig,
+		ClusterName: "broker",
+	}
```

Comment:
You should make a complete copy of `info`,  ie `brokerInfo := *info`. However I suggest just making this implicit in the function signature, ie pass `info` by value rather than pointer so each invocation gets its own copy and can modify it as they see fit:  

```suggestion
	info.RestConfig = brokerRestConfig
	info.ClusterName: = "broker"
	
	info.DynClient, info.ClientSet, err = getClients(brokerRestConfig)
	...
```